src/redux/sagas/utils/proofs.ts

```diff
@@ -196,52 +206,124 @@ const getMultiPermissionProofsLocal = async ({
+      const proof = await getMultiPermissionProofsLocal({
+        networkClient,
+        colonyRoles,
+        colonyDomains,
+        requiredDomainId,
+        requiredColonyRoles: roleGroup,
+        permissionAddress,
+        isMultiSig,
+      });
+
+      // If we get here, it means the current roleGroup satisfies the conditions,
+      // so we can return its proof.
+      return proof;
+    } catch (error) {
+      // Catch any errors and continue to the next RoleGroup in the RoleGroupSet
+      // No need to do anything here, just continue to the next iteration
+    }
+  }
+
+  // If we exhaust all RoleGroups in the RoleGroupSet without finding a valid proof,
+  // we throw an error indicating none of the RoleGroups have the required permissions.
+  throw new Error(
+    `${permissionAddress} does not have any required roles in any of the specified domains`,
+  );
+};
+
 interface GetPermissionProofsLocalParams {
   networkClient: ColonyNetworkClient;
   colonyRoles: ColonyRoleFragment[];
   colonyDomains: Domain[];
   requiredDomainId: number;
-  requiredColonyRole: ColonyRole | ColonyRole[];
+  requiredColonyRoles: RequiredColonyRoleGroup;
   permissionAddress: string;
   isMultiSig?: boolean;
 }
+
 export const getPermissionProofsLocal = async ({
   networkClient,
   colonyRoles,
   colonyDomains,
   requiredDomainId,
-  requiredColonyRole,
+  requiredColonyRoles,
   permissionAddress,
   isMultiSig = false,
 }: GetPermissionProofsLocalParams): Promise<[BigNumber, BigNumber, string]> => {
-  if (Array.isArray(requiredColonyRole)) {
-    if (requiredColonyRole.length === 1) {
+  if (!Array.isArray(requiredColonyRoles)) {
+    const singleRole = requiredColonyRoles;
+    // ColonyRole.Root
+    return getSinglePermissionProofsLocal({
+      networkClient,
+      colonyRoles,
+      colonyDomains,
+      requiredDomainId,
+      requiredColonyRole: singleRole,
+      permissionAddress,
+      isMultiSig,
+    });
+  }
+
+  if (!Array.isArray(requiredColonyRoles[0])) {
+    const roleGroup = requiredColonyRoles as RoleGroup;
+    if (roleGroup.length === 1) {
+      // [ColonyRole.Root]
       return getSinglePermissionProofsLocal({
         networkClient,
         colonyRoles,
         colonyDomains,
         requiredDomainId,
-        requiredColonyRole: requiredColonyRole[0],
+        requiredColonyRole: roleGroup[0],
         permissionAddress,
         isMultiSig,
       });
     }
+    // [ColonyRole.Root, ColonyRole.Architecture]
     return getMultiPermissionProofsLocal({
       networkClient,
       colonyRoles,
       colonyDomains,
       requiredDomainId,
-      requiredColonyRoles: requiredColonyRole,
+      requiredColonyRoles: roleGroup,
       permissionAddress,
       isMultiSig,
     });
   }
-  return getSinglePermissionProofsLocal({
+  const roleGroupSet = requiredColonyRoles as RoleGroupSet;
+  // [[ColonyRole.Root]] // [[ColonyRole.Root, ColonyRole.Architecture]] // [[ColonyRole.Root], [ColonyRole.Architecture]]
```

Comment:
Leftover ?

Reply:
Intentional to show the different permission sets that we are expecting to get through to this part of the function. I should probably have made the comment clearer though!